Select position of ThermalAssistant

webbsimax opened the pull request:
Allow user to select position of ThermalAssistant.
Includes avoiding infoboxes if in the way.


<!--

Thank you for your interest in contributing to XCSoar! Please read the
following information to make it easier for us to review your changes.

We appreciate if you make sure to:

  - Document the changes (in the NEWS.txt file, and the manual when relevant)
  - Enable maintainer edits[1] (in case we need to help with the PR)
  - Check your commits and their messages (so they're all tidy and coherent)

[1] https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

-->


Brief summary of the changes
----------------------------

<!--
Please note that the commit messages is where detailed descriptions of the
changes should be made - this section is just for a brief summary/overview.
-->


Related issues and discussions
------------------------------

<!--
Please link any relevant issues or forum posts here, for reference.

If this PR resolves an existing issue, please write "Closes #1234" so that
the issue is closed automatically when this PR is merged.
-->

MaxKellermann commented:
Don't close PRs and reopen new ones. This makes it harder to review your code.

webbsimax commented:
I'm not sure how the previous one closed - if it's possible to reopen let me know and I'll do that.

MaxKellermann commented on src/UISettings.hpp:
```diff
@@ -52,8 +52,17 @@ struct UISettings {
-  /** Show ThermalAssistant if circling */
-  bool enable_thermal_assistant_gauge;
+  /** Show ThermalAssistant if circling (obsolete - use thermal_assistant_position)*/
+  bool enable_thermal_assistant_gauge_obsolete;
```
Nobody uses this variable, so what's the point of storing it?

webbsimax replied:
Used in UIProfile.cpp to to migrate the old setting, as required in the previous PR

webbsimax replied:
Have moved it to a local variable in UIProfile and rebased.

MaxKellermann commented:
> I'm not sure how the previous one closed - if it's possible to reopen let me know and I'll do that.

It was you who closed it: "webbsimax closed this 11 minutes ago"

webbsimax commented:
Anything else required? In addition to the feedback above, from the previous PR I have:

- addressed whitespace issue
- removed "requires restart", replaced with reload of mainwindow
- fixed indent issue
- sorted out the unnecessary default case
- resolve the lack of migration of old settings ThermalAssistant.

MaxKellermann commented:
You messed up your commit. Look at the diff.
Hint: always check the diffs before submitting.

webbsimax commented:
Not sure what happened there, rebasing ended up pulling some other commits in. 
Should be fixed and all in one commit now.

MaxKellermann commented:
You added lots of trailing space, please fix those.

webbsimax commented:
> You added lots of trailing space, please fix those.

Done
